### ISSUE TITLE
Missing query parameters from traits

### ISSUE THREAD

**Issue by dc950** [raml-org/raml-java-parser#411]
I've come across an issue (using version `1.0.12`) where query parameters from traits are not present in the model under certain conditions.

It seems to be related to the parameters being defined as not required using a `?` and the order of the traits in the `is`.

With the following raml spec:

```
#%RAML 1.0
title: test
baseUri: /test

traits:
  trait1:
    queryParameters:
      t1?:
        type: string
  trait2:
    queryParameters:
      t2:
        type: string
        required: false

/test:
  get:
    is: [trait1, trait2]
  put:
    is: [trait2, trait1]
```

And the following java code just to look at the query parameters:

```
class Main {
  public static void main(String[] args) {
    RamlModelResult ramlModelResult = new RamlModelBuilder().buildApi("test.raml");
    Api api = ramlModelResult.getApiV10();
    for(Resource res : api.resources()) {
      for(Method method : res.methods()) {
        System.out.println("\nMethod: " + method.method());
        System.out.println("Number of query parameters: " + method.queryParameters().size());
        for(TypeDeclaration param : method.queryParameters()) {
          System.out.println("  - " + param.name());
        }
      }
    }
  }
}
```
The output is this:
```
Method: get
Number of query parameters: 2
  - t1
  - t2

Method: put
Number of query parameters: 1
  - t2
```
Also, setting `t2:` to be `t2?:`  and removing the `required: false` makes both methods contain only 1 of the parameters:

```
Method: get
Number of query parameters: 1
  - t1

Method: put
Number of query parameters: 1
  - t2
```

**Comment by kevinduffey** [comment]
This is a year old.. any fix for this? I am using RAML2JAXRS and am seeing this issue.. or something similar. I have an uber trait that IS: multiple traits. For some reason, raml2html and Atom IDE see/render them fine. But RAML2JAXRS some how ends up completely ignoring them.. producing a method with no parameters or response types. In my traits, I have the uber trait made up of 2 or more traits each of which return a specific response code.

I am guessing the issue RAML2JAXRS is seeing is that there are no traits at all from the parser. It is as if it ignores the combined trait at the parser.

**Comment by stojsavljevic** [comment]
Hi,

I don't find this issue resolved in `1.0.27`.
I'm getting the very same behavior for example above.

Here is another example:
```
#%RAML 1.0
title: Missing query parameters from traits #411

traits:
    metaQuery:
      queryParameters:
        color: string
        brand-name: string
    
    paginationParameters:
      queryParameters:
        begin-index?:
          type: integer
        max-elements?:
          type: integer

/bus:
  get:
    is: [metaQuery, paginationParameters]
    queryParameters:
      brandN:
        type: string
        enum: [Volvo, Scania]
    responses:
      200:
        body:
          application/json:
            type: string
```
The output is this:
```
Method: get
Number of query parameters: 3
  - brandN
  - color
  - brand-name
```
`begin-index` and `max-elements` are ignored.

**Comment by dfmacias** [comment]
Hi Aleksandar,

You're right, that fix introduced a regression bug, so we had to remove it from the release. It will be fixed in 1.0.28.

I already updated the milestone. Sorry for the confusion
